datafusion/core/src/physical_planner.rs

```diff
@@ -590,6 +593,58 @@ impl DefaultPhysicalPlanner {
                     );
                 }
             }
+            LogicalPlan::Dml(DmlStatement {
+                table_name,
+                target,
+                op: WriteOp::Delete,
+                input,
+                ..
+            }) => {
+                if let Some(provider) =
+                    target.as_any().downcast_ref::<DefaultTableSource>()
+                {
+                    let filters = extract_dml_filters(input)?;
```

Comment:
I think a  better (smaller API surface) would be to simply try and call `provider.delete_from(session_state, filters)` and pass along the context

Then you can remove `DmlCapabilities` entirely